### ISSUE TITLE
Instance variable with same name as class erroneously flagged as unbound

### ISSUE THREAD

**Issue by Godsmith** [microsoft/pylance-release#175]
## Environment data

-   Language Server version: 2020.7.3
-   OS and version: Windows 10 1909
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8

## Expected behaviour

When I have a module with the following code:

```
class bar:
    pass


class Foo:
    def __init__(self, bar_: bar) -> None:
        self.bar = "hello"
```

no problems shall be listed.

## Actual behaviour

A problem appears in the Problems tab, and squiggly lines are shown under the type of the `bar_` argument:

`"bar" is unbound Pylance (reportUnboundVariable)`

## Logs

```
[BG] analyzing: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (5ms)
[FG] parsing: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (0ms)
[FG] binding: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (0ms)
[BG] analyzing: c:\Users\filip\workspace\timereporter\timereporter\datetest.py ...
[BG]   parsing: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (4ms)
[BG]   binding: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (1ms)
[BG]   checking: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (27ms)
[BG] analyzing: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (33ms)
[FG] parsing: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (1ms)
[FG] binding: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (0ms)
[BG] analyzing: c:\Users\filip\workspace\timereporter\timereporter\datetest.py ...
[BG]   parsing: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (1ms)
[BG]   binding: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (0ms)
[BG]   checking: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (8ms)
[BG] analyzing: c:\Users\filip\workspace\timereporter\timereporter\datetest.py (9ms)
```

## Code Snippet / Additional information

One might think it would be rare that you would have a variable with the same name as a class, however, some packages like `datetime` has lowercase class names. In fact, I found this bug when I tried to name a variable `date` while I had imported the `date` class from `datetime`.

**Comment by erictraut** [comment]
Thanks for the bug report. I just introduced a fix for this problem. It will be addressed in the next version of Pylance.

**Comment by CatNofishing** [comment]
Good!

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.7.4, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202074-29-july-2020
